main/src/ca/uwaterloo/flix/language/phase/unification/BoolFormula.scala

```diff
@@ -155,6 +155,7 @@ object BoolFormula {
     case Type.Apply(Type.Cst(TypeConstructor.Complement, _), tpe1, _) => Neg(fromEffType(tpe1, m))
     case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Intersection, _), tpe1, _), tpe2, _) => Conj(fromEffType(tpe1, m), fromEffType(tpe2, m))
     case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Union, _), tpe1, _), tpe2, _) => Disj(fromEffType(tpe1, m), fromEffType(tpe2, m))
+    case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Difference, _), tpe1, _), tpe2, _) => Conj(fromEffType(tpe1, m), Neg(fromEffType(tpe2, m)))
```

Comment:
Is the plan to get rid of this or?

Reply:
By keeping it aren't we allowing sugar to leak into the types?

Reply:
Sure. I just didn't want a user's written signature to be unrecognizable after typing. But if we add the resugaring for displaying differences, it should be OK